src/OptionsCompounderAave2.sol

```diff
@@ -271,57 +223,91 @@ contract OptionsCompounder is FlashLoanReceiverBase, Ownable {
+        );
+        console2.log(
+            "2.Balance of wantToken: ",
+            IERC20(wantToken).balanceOf(address(this))
         );
         console2.log(
-            "2.Balance of asset: ",
-            IERC20(asset).balanceOf(address(this))
+            "2.Balance of paymentToken: ",
+            paymentToken.balanceOf(address(this))
         );
-        /* Asset and paymentToken are the same addresses */
+
         /* Repay the debt and revert if it is not profitable */
         uint256 assetBalance = paymentToken.balanceOf(address(this));
-        console2.log("2.Balance of paymentToken: ", assetBalance);
-        console2.log("2.Amount to pay back: ", totalAmount);
 
-        if ((assetBalance - initialBalance) < totalAmount) {
+        if ((assetBalance - initialBalance) <= totalAmount) {
             revert OptionsCompounder__FlashloanNotProfitable(
                 (assetBalance - initialBalance),
                 totalAmount
             );
         }
-        _gain = (assetBalance - initialBalance) - totalAmount; // Question: still we shall use safe math ?
-        senderToBalance[sender] = _gain;
-        IERC20(asset).approve(address(LENDING_POOL), totalAmount);
+        /* Protected by statement above */
+        gainInPaymentToken = (assetBalance - initialBalance) - totalAmount;
+
+        /* Approve the underlying token to make swap */
+        IERC20(asset).approve(address(swapperSwaps), gainInPaymentToken);
+
+        /* Transfer gain to the sender */
+        swapperSwaps.swapBal(
+            asset,
+            wantToken,
+            gainInPaymentToken,
+            minAmountOutData,
+            BEETX_VAULT_OP
+        );
+
+        gainInWantToken = IERC20(wantToken).balanceOf(address(this));
 
-        return _gain;
+        /* Approve lending pool to spend borrowed tokens + premium */
+        IERC20(asset).approve(address(lendingPool), totalAmount);
+
+        /* Console log area - temporary */
+        console2.log(
+            "3.Balance of underlyingToken: ",
+            IERC20(underlyingToken).balanceOf(address(this))
+        );
+        console2.log("3.Balance of wantToken: ", gainInWantToken);
+        console2.log(
+            "3.Balance of paymentToken: ",
+            paymentToken.balanceOf(address(this))
+        );
+        console2.log("3.Amount paid back: ", totalAmount);
+
+        return gainInWantToken;
     }
 
     /***************************** Getters ***********************************/
-    function isStrategyAdded(address _strategy) public view returns (bool) {
-        address[] memory tmpStrategies = strategies;
-        bool strategyFound = false;
-        for (uint8 idx = 0; idx < tmpStrategies.length; idx++) {
-            if (tmpStrategies[idx] == _strategy) {
-                strategyFound = true;
-                break;
-            }
-        }
-        return strategyFound;
+    /* Temporary for testing - apy will be caluclated in vault */
```

Comment:
just to clarify its not rly calculated in the vault either its mostly offchain stuff now